Blockzilla/Modules/WebView/WebViewController.swift

```diff
@@ -62,6 +62,7 @@ class WebViewController: UIViewController, WebController {
 
     var printFormatter: UIPrintFormatter { return browserView.viewPrintFormatter() }
     var scrollView: UIScrollView { return browserView.scrollView }
+    private let scriptHandlers = (tpPreload: "focusTrackingProtection", tpPostLoad: "focusTrackingProtectionPostLoad", findInPage: "findInPageHandler")
```

Comment:
This would probably be better as a String enum. Are we using Swift 4.2? We might be able to do this with that as well :) https://github.com/apple/swift-evolution/blob/master/proposals/0194-derived-collection-of-enum-cases.md

Reply:
Can you explain why an enum is better here?

Reply:
We're also not on Swift 4.2 yet (that won't be until the iOS 12 release)

Reply:
@oliviabrown9 I'm so out of the loop with Swift releases! I need to catch back up!

Regarding enums vs tuples: I think both methods are perfectly valid and can do the job. I think where an enum is better is being able to leverage language semantics. It gives you a Type that defines a finite set of options and gives you the ability to add behavior to it (for example: Adding the `allValues` property. Whenever you add another case to the enum you would won't have to remember to also add the case to the array to loop over them).

Another benefit that we aren't using here is it gives us a Type we can use in method signatures so we don't have to rely directly on strings.

for example: 

```swift
private extension WKUserContentController {
    func add(_ scriptMessageHandler: WKScriptMessageHandler, handler: ScriptHandler) {
        add(scriptMessageHandler, name: handler.rawValue)
    }
}
```

So when I come back to the code it becomes easier to remember what the type is for, and what I can do with it.